tests/e2e/monitoring/test_monitoring_when_osd_down.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import constants, ocp, defaults
+from ocs_ci.framework.testlib import tier4, E2ETest
+from tests.sanity_helpers import Sanity
+from ocs_ci.ocs.monitoring import check_pvcdata_collected_on_prometheus
+from ocs_ci.ocs.resources import pod
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.polarion_id("OCS-605")
+@tier4
+class TestMonitoringWhenOSDDown(E2ETest):
+    """
+    When one of the osd pod is down, there shouldn't be any functional impact on prometheus pod
+    and also all data/metrics should be collected correctly.
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def test_fixture(self, pod_factory, num_of_pod=3):
+        """
+        Create resources for the test
+        """
+        self.pod_objs = [
+            pod_factory(
+                interface=constants.CEPHBLOCKPOOL,
+                status=constants.STATUS_RUNNING
+            ) for _ in range(num_of_pod)
+        ]
+
+        # Check for the created pvc metrics on prometheus pod
+        for pod_obj in self.pod_objs:
+            assert check_pvcdata_collected_on_prometheus(pod_obj.pvc.name), (
+                f"On prometheus pod for created pvc {pod_obj.pvc.name} related data is not collected"
+            )
+
+    def test_monitoring_when_osd_down(self, pod_factory):
+        """
+        Test case to validate monitoring when osd is down
+        """
+
+        # Get osd pods
+        osd_pod_list = pod.get_osd_pods()
+
+        # Make one of the osd down(first one)
+        ocp_obj = ocp.OCP(kind=constants.DEPLOYMENT, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+
+        params = {'spec': {'replicas': 0}}
+        name = osd_pod_list[0].get().get('metadata').get('name')
+        assert ocp_obj.patch(resource_name=name[:-17], params=params), (
```

Comment:
Please don't get deployment name as `name[:-17]`. It might work now but if the name of deployment or pod changes in future then this will break and it will be a pain to debug it. You can get name of first osd deployment like:
```
    osd_deployments = ocp_obj.get(selector=constants.OSD_APP_LABEL).get('items')
    name = osd_deployments[0].get('metadata').get('name')
```

Reply:
Here, name will have `rook-ceph-osd-0-9bfb588db-mjn4q` but I need only `rook-ceph-osd-0` as resource_name.